Model/Api/OAuthRedirect.php

```diff
@@ -474,7 +502,47 @@ private function linkLoginAndRedirect($externalID, $customerID)
             $this->getCookieManager()->deleteCookie('mage-cache-sessid', $metadata);
         }
 
+        try {
+            // we should force to set new "form_key" cookie here to prevent safari "session expired" issue
+            // when we are logged-in by SSO without page redirect (by using Bolt Shopper Assistant for example)
+            // in this case magento removing cookie by backend here:
+            // https://github.com/magento/magento2/blob/5844ade68b2f9632e3888c81c946068eba6328bb/app/code/Magento/PageCache/Observer/FlushFormKey.php
+            // but on safari without redirect the cookie is removed from browser (you can't see cookie on the devtool) but not in window.document object
+            // (we can see the cookie "form_key" there by checking object in browser console: "document.cookie")
+            // therefore all future requests is being called without "form_key" which is the reason of  "session expire" issue.
+            // By normal flow Magento updates that cookie in file:
+            // 2.3.0 https://github.com/magento/magento2/blob/44a7b6079bcac5ba92040b16f4f74024b4f34d09/app/code/Magento/PageCache/view/frontend/web/js/page-cache.js#L115
```

Comment:
The version below 2.4.0 affected, correct? By the way, what about 2.2.x, we did not drop it's support yet.

Reply:
updated comment, under 2.2.x the same code